### ISSUE TITLE
Support "image" source type

### ISSUE THREAD

**Issue by tmcw** [mapbox/mapbox-gl-native#1350]
Much like #601 this is a parity issue. V8 can ship without this https://github.com/mapbox/mapbox-gl-style-spec/pull/264 but we want to implement Video & Image source support in gl-native.

Description: like video source, except it doesn't move and it's an image.

**Comment by friedbunny** [comment]
For those attempting to use styles with image sources, this is the error they'll see:

```
{Map}[Style]: Failed to parse [image.png]: 0 - Invalid value.
```

**Comment by damianw** [comment]
Is any progress being made on this + #4311 (what is the priority for a future release)?

**Comment by incanus** [comment]
Here's a workaround using the annotation view API that's pretty close for most use cases: 

https://gist.github.com/incanus/7e19ca99f76c0b3d32ae147872325722

<a href="https://dl.dropboxusercontent.com/u/575564/overlay.mp4"><img src="https://dl.dropboxusercontent.com/u/575564/overlay.png"/></a>

**Comment by aayers-agribile** [comment]
Is support for this feature planned for the iOS SDK?

**Comment by alayers2** [comment]
Is support for this feature planned for the iOS SDK?

**Comment by aottenwess** [comment]
Also curious if there is a plan for the iOS SDK

**Comment by boundsj** [comment]
This issue is tracking implementation in `Core GL` which means the underlying C++ implementation of this library. The platform specific (i.e. iOS, Android, Qt) wrappers can be completed after the core work is done. At this time, this feature is not on any platform release milestones and our focus is on completing the implementation of data-driven styling in core and all platforms.

**Comment by damianw** [comment]
Honestly, that's ridiculous. It's been over a year since your team has dropped support for the legacy SDK. Not only that, but you've _deleted_ the repo for it -- which is unheard of. And yet, all this time later, you're saying that achieving _feature parity_ with the old SDK is still _low priority_? I suspect we are not your only customer that is getting increasingly agitated that your team has provided neither a suitable replacement for the features which have been yanked from under their feet nor support for the old tools that you provided.

**Comment by incanus** [comment]
Give this a try, possible today with the 3.5.0-series data-driven styling betas, and see if it meets your use cases. 

**tl;dr** You can do this yourself with the capabilities that we **are** focusing on in the SDK... i.e. more general-use functionality allowing you to accomplish _many kinds of different things_ and better building a long-term foundation for our tools. 

https://gist.github.com/incanus/8c277a0700c917ac58600d93cd849bb8

![overlay](https://cloud.githubusercontent.com/assets/17722/23531467/657eb8ca-ff5b-11e6-81f2-ab3305b17c9c.gif)

**Comment by incanus** [comment]
BTW the above is more performant than the other workaround detailed in https://github.com/mapbox/mapbox-gl-native/issues/1350#issuecomment-261358599.

**Comment by damianw** [comment]
Thanks for the workaround; unfortunately I can't make use of it since I am waiting for the feature on Android, but I will send it to our iOS team to see if it works well enough for them to update to the new SDK. Sorry if that wasn't clear. Are you aware of similar workarounds for Android?

**Comment by incanus** [comment]
Yes, this will work just the same on Android, as data-driven styling is supported there as well. See the `GeoJsonSource`, `SymbolLayer`, and `CameraFunction` classes to replicate the source, layer, and dynamic sizing functionality as in the iOS example. You'll need the beta there, too, which is 5.0.0-beta.1 or -beta.2.

**Comment by aayers-agribile** [comment]
In the previous workaround for this issue the image is an annotation, which means we can attach uitouchevents to it. Is there any way to attach touch events or trigger actions from interaction with the images with the data-driven styling solution?

**Comment by aayers-agribile** [comment]
This workaround is really good. Using the data-driven styling makes everything a lot easier. My only follow-up question that I wasn't able to find in the docs is that after adding ~15 images with this workaround, the layers stop showing up. Is there a limit on layers that can be added to a style in the iOS SDK?

**Comment by alayers2** [comment]
This workaround is really good. I was able to test it today and it looks way better than using the UIView based annotations and shoving images into them. The only problem with this approach is that the spritemap fills up very quickly when working with images of any size, and I can't add all of the images that I want to the map. (At least I assume that's what is happening) Is there any way to work around that limitation besides increasing the size of where the map stores images for a style?

**Comment by jamala** [comment]
In using this workaround to show multiple images, using a symbol layer for each image. However, only 12 images are visible on the map. Is this the same issue as described in the above comment?

**Comment by tikimcfee** [comment]
> Also, are there plans to support images as a first-class feature instead of using the symbol layer as a workaround?

Hey! Giving a big +1 to this, as the symbol-layer solution (though functional) isn't really ideal. You've got the same marker-drag-lagginess issue which looks wonky, and there's a lot more overheard to handle actually adding these things to client maps.

**Comment by jfalkner** [comment]
Echoing the above. Need support for this on Android, and the work-around isn't a great fix.

**Comment by 1ec5** [comment]
This feature is being implemented in #8968 and #9110.

**Comment by friedbunny** [comment]
Image sources are now available in iOS v3.7.0 (e.g., [`MGLImageSource`](https://www.mapbox.com/ios-sdk/api/3.7.0/Classes/MGLImageSource.html)) and Android v5.2.0.
